nni/algorithms/compression/v2/pytorch/pruning/movement_pruner.py

```diff
@@ -140,16 +148,34 @@ class MovementPruner(EvaluatorBasedPruner):
         {evaluator_docstring}
         The old API (``trainer``, ``traced_optimizer`` and ``criterion``) is still supported and will be deprecated in v3.0.
         If you want to consult the old API, please refer to `v2.8 pruner API <https://nni.readthedocs.io/en/v2.8/reference/compression/pruner.html>`__.
-    training_epochs : int
+    training_epochs
         The total epoch number for training the model.
         Make sure the total `optimizer.step()` in `training_epochs` is bigger than `cool_down_beginning_step`.
-    warm_up_step : int
+    warm_up_step
         The total `optimizer.step()` number before start pruning for warm up.
         Make sure `warm_up_step` is smaller than `cool_down_beginning_step`.
-    cool_down_beginning_step: int
+    cool_down_beginning_step
         The number of steps at which sparsity stops growing, note that the sparsity stop growing doesn't mean masks not changed.
         The sparsity after each `optimizer.step()` is:
         total_sparsity * (1 - (1 - (current_step - warm_up_step) / (cool_down_beginning_step - warm_up_step)) ** 3).
+    regular_scale
+        Use to scale the movement score regular loss. In 'soft' mode, higher regular scale means higher final sparsity.
+        The recommended range is 1 ~ 30.
+    movement_mode
+        'hard' or 'soft'. Note that in 'soft' mode, ``sparsity`` set in the ``config_list`` means the sparsify threshold,
+        'soft' mode cannot precisely control the sparsity rate, but usually has higher performance compared with 'hard' mode.
+        ``sparsity`` in 'soft' mode usually set to ``0.1``, and using ``regular_scale`` to control the final relative sparsity.
+
+        For detailed differences between 'hard' and 'soft', please refer to the paper.
+        In short, 'hard' means that the corresponding layer is pruned to a fixed ratio by the topk method according to the movement score,
+        which is the sparsity ratio set in config_list.
+        'soft' means that the final sparsity size will not be fixed, but the generation of the mask will be controlled by a threshold,
+        and the positions corresponding to scores below the threshold will be masked during the movement training process.
+    sparse_granularity
+        This is an experimental interface, by default, apply 'finegrained' pruning. If 'auto' is set, will try to apply structure pruning.
+        For the attention layer, will apply block sparse with size [head_width, head_width]. For the following two linear layers (FFN),
```

Comment:
what is the meaning of "head_width"?

Reply:
head_width is width of the output of one attention head